### PR TITLE
HOCS-3970 Fix case creation for cases created by case-creator

### PR DESCRIPTION
**Pull request by robertdeniszczyc2**
Ensures case-creator can create cases and removes mutation of data to improve readability.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java [hunk 1]
```diff
@@ -69,53 +69,48 @@ public WorkflowService(CaseworkClient caseworkClient,
 
     public CreateCaseResponse createCase(String caseDataType, LocalDate dateReceived, List<DocumentSummary> documents, UUID userUUID, UUID fromCaseUUID, Map<String, String> receivedData) {
         // Create a case in the casework service in order to get a reference back to display to the user.
-        Map<String, String> data = new HashMap<>(receivedData);
         CreateCaseworkCorrespondentRequest correspondentRequest = null;
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:72]: Defaults to `null` so not needed.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java [hunk 2]
```diff
@@ -69,53 +69,48 @@ public WorkflowService(CaseworkClient caseworkClient,
...
-                    .address1(data.get("Address1"))
-                    .address2(data.get("Address2"))
-                    .address3(data.get("Address3"))
-                    .country(data.get("Country"))
-                    .telephone(data.get("Telephone"))
-                    .email(data.get("Email"))
-                    .reference(data.get("Reference"))
-                    .build();
-
-            data.remove("Fullname");
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:91]: Wow, glad to see this going 👍🏽

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java (hunk 1 above)
Comment (robertdeniszczyc2) [reply src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:72]: If the assign to null is removed the null check on line 115 warns that the variable might not have been initialised.

**Comment by robertdeniszczyc2** [comment]
@LiamMacP are you happy with me Resolving the comments added please? Cheers

**Comment by LiamMacP** [comment]
Yep fine with me, something to revisit in the future...

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowConstants.java [hunk 3]
```diff
@@ -7,4 +7,20 @@
     String LAST_UPDATED_BY_USER = "LastUpdatedByUserUUID";
     String DATE_RECEIVED = "DateReceived";
     String CASE_REFERENCE = "CaseReference";
+    String FULL_NAME = "Fullname";
```
[the remark this answers is not in the document]
Comment (robertdeniszczyc2) [reply src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowConstants.java:10]: final is redundant on interface fields so not required

**Pull request merged**